WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -571,4 +571,101 @@
+    let!(:organizer) { FactoryGirl.create(:user) }
+    let!(:future_competition1) { FactoryGirl.create(:competition, :registration_open, starts: 3.week.from_now, organizers: [organizer], delegates: [delegate], eventSpecs: "222 333") }
+    let!(:future_competition2) { FactoryGirl.create(:competition, :registration_open, starts: 2.weeks.from_now, organizers: [organizer], eventSpecs: "222 333") }
+    let!(:future_competition3) { FactoryGirl.create(:competition, :registration_open, starts: 1.weeks.from_now, organizers: [organizer], eventSpecs: "222 333") }
+    let!(:past_competition1) { FactoryGirl.create(:competition, :registration_open, starts: 1.month.ago, organizers: [organizer], eventSpecs: "222 333") }
+    let!(:past_competition2) { FactoryGirl.create(:competition, starts: 2.month.ago, delegates: [delegate], eventSpecs: "222 333") }
+    let!(:past_competition3) { FactoryGirl.create(:competition, :registration_open, starts: 3.month.ago, delegates: [delegate], eventSpecs: "222 333") }
+    let!(:unscheduled_competition1) { FactoryGirl.create(:competition, delegates: [delegate], eventSpecs: "222 333", year: "0") }
+    let!(:registered_user) { FactoryGirl.create :user, name: "Jan-Ove Waldner" }
+    let!(:registration1) { FactoryGirl.create(:registration, competitionId: future_competition1.id, user: registered_user) }
+    let!(:registration2) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: registered_user) }
+    let!(:registration3) { FactoryGirl.create(:registration, competitionId: past_competition1.id, user: registered_user) }
+    let!(:registration4) { FactoryGirl.create(:registration, competitionId: past_competition3.id, user: organizer) }
+    let!(:registration5) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: delegate) }
+    let!(:results_person) { FactoryGirl.create(:person, id: "2014PLUM01", name: "Jeff Plumb") }
+    let!(:results_user) { FactoryGirl.create :user, name: "Jeff Plumb", wca_id: "2014PLUM01" }
+
+    context 'when not signed in' do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :my_competitions
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as user with results for a comp they did not register for' do
+      before do
+        sign_in results_user
+      end
+
+      it 'shows my upcoming and past competitions' do
+        Result.create!(
+          pos: 1,
+          personId: "2014PLUM01",
+          personName: "Jeff Plumb",
+          countryId: "AUS",
+          competitionId: past_competition1.id,
+          eventId: "333",
+          roundId: "f",
+          formatId: "a",
+          value1: 2000,
+          value2: 2000,
+          value3: 2000,
+          value4: 2000,
+          value5: 2000,
+          best: 2000,
+          average: 2000,
+          regionalSingleRecord: "WR",
+          regionalAverageRecord: "WR",
+        )
```

Comment:
Good idea, I thought I tried something similar but must have had a typo somewhere. I'll update this.